flink-test-utils-parent/flink-test-utils/src/test/java/org/apache/flink/streaming/util/PseudoRandomValueSelectorTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeNoException;
+import static org.junit.Assume.assumeNotNull;
+
+/** Tests {@link PseudoRandomValueSelector}. */
+public class PseudoRandomValueSelectorTest {
+
+    /**
+     * Tests that the selector will return different values if invoked several times even for the
+     * same option.
+     */
+    @Test
+    public void testRandomizationOfValues() {
+        final Duration[] alternatives =
+                IntStream.range(0, 1000).boxed().map(Duration::ofMillis).toArray(Duration[]::new);
+
+        final PseudoRandomValueSelector valueSelector = PseudoRandomValueSelector.create("test");
+
+        final Set<Duration> uniqueValues = new HashSet<>(1);
+        for (int i = 0; i < 100; i++) {
+            final Duration selectedValue =
+                    selectValue(valueSelector, CHECKPOINTING_INTERVAL, alternatives);
+            uniqueValues.add(selectedValue);
+        }
+        assertNotEquals(1, uniqueValues.size());
+    }
+
+    @Nonnull
+    private <T> T selectValue(
+            PseudoRandomValueSelector valueSelector, ConfigOption<T> option, T... alternatives) {
+        final Configuration configuration = new Configuration();
+        assertNull(configuration.get(option));
+        valueSelector.select(configuration, option, alternatives);
+        final T selected = configuration.get(option);
+        assertNotNull(selected);
+        return selected;
+    }
+
+    /** Tests that the selector will return different values for different seeds. */
+    @Test
+    public void testRandomizationWithSeed() {
+        final Duration[] alternatives =
+                IntStream.range(0, 1000).boxed().map(Duration::ofMillis).toArray(Duration[]::new);
+
+        final Set<Duration> uniqueValues = new HashSet<>(1);
+        for (int i = 0; i < 100; i++) {
+            final PseudoRandomValueSelector selector = PseudoRandomValueSelector.create("test" + i);
+            uniqueValues.add(selectValue(selector, CHECKPOINTING_INTERVAL, alternatives));
+        }
+        assertNotEquals(1, uniqueValues.size());
+    }
+
+    /** Tests that it produces the same value for the same seed. */
+    @Test
+    public void testStableRandomization() {
+        final Duration[] alternatives =
+                IntStream.range(0, 1000).boxed().map(Duration::ofMillis).toArray(Duration[]::new);
+
+        final Set<Duration> uniqueValues = new HashSet<>(1);
+        for (int i = 0; i < 100; i++) {
+            final PseudoRandomValueSelector selector = PseudoRandomValueSelector.create("test");
+            uniqueValues.add(selectValue(selector, CHECKPOINTING_INTERVAL, alternatives));
```

Comment:
maybe add another set for the second value, to test that a sequence of selections is also deterministic.

Reply:
I'm checking a triple of config options now.